crates/fuel-core/src/graphql_api/ports.rs

```diff
@@ -387,55 +387,144 @@ pub mod worker {
+            evictor_cache::DaCompressionTemporalRegistryEvictorCacheV2,
+            predicate_code::DaCompressionTemporalRegistryPredicateCodeV2,
+            registry_index::DaCompressionTemporalRegistryIndexV2,
+            script_code::DaCompressionTemporalRegistryScriptCodeV2,
+            timestamps::DaCompressionTemporalRegistryTimestampsV2,
+        };
+
+        pub trait OffChainDatabaseTransaction: StorageMutate<OwnedMessageIds, Error = StorageError>
+            + StorageMutate<OwnedCoins, Error = StorageError>
+            + StorageMutate<FuelBlockIdsToHeights, Error = StorageError>
+            + StorageMutate<ContractsInfo, Error = StorageError>
+            + StorageMutate<OldFuelBlocks, Error = StorageError>
+            + StorageMutate<OldFuelBlockConsensus, Error = StorageError>
+            + StorageMutate<OldTransactions, Error = StorageError>
+            + StorageMutate<SpentMessages, Error = StorageError>
+            + StorageMutate<RelayedTransactionStatuses, Error = StorageError>
+            + StorageMutate<CoinBalances, Error = StorageError>
+            + StorageMutate<MessageBalances, Error = StorageError>
+            + StorageMutate<CoinsToSpendIndex, Error = StorageError>
+            + StorageMutate<DaCompressedBlocks, Error = StorageError>
+            + StorageMutate<DaCompressionTemporalRegistryAddress, Error = StorageError>
+            + StorageMutate<DaCompressionTemporalRegistryAssetId, Error = StorageError>
+            + StorageMutate<DaCompressionTemporalRegistryContractId, Error = StorageError>
+            + StorageMutate<DaCompressionTemporalRegistryScriptCode, Error = StorageError>
+            + StorageMutate<
+                DaCompressionTemporalRegistryPredicateCode,
+                Error = StorageError,
+            > + StorageMutate<DaCompressionTemporalRegistryIndex, Error = StorageError>
+            + StorageMutate<DaCompressionTemporalRegistryTimestamps, Error = StorageError>
+            + StorageMutate<DaCompressionTemporalRegistryEvictorCache, Error = StorageError>
+            + StorageMutate<AssetsInfo, Error = StorageError>
+            + StorageMutate<DaCompressionTemporalRegistryAddressV2, Error = StorageError>
+            + StorageMutate<DaCompressionTemporalRegistryAssetIdV2, Error = StorageError>
+            + StorageMutate<DaCompressionTemporalRegistryContractIdV2, Error = StorageError>
+            + StorageMutate<DaCompressionTemporalRegistryScriptCodeV2, Error = StorageError>
+            + StorageMutate<
+                DaCompressionTemporalRegistryPredicateCodeV2,
+                Error = StorageError,
+            > + StorageMutate<DaCompressionTemporalRegistryIndexV2, Error = StorageError>
+            + StorageMutate<DaCompressionTemporalRegistryTimestampsV2, Error = StorageError>
+            + StorageMutate<
+                DaCompressionTemporalRegistryEvictorCacheV2,
+                Error = StorageError,
+            >
```

Comment:
Since these are the only one that has changed, we could gather these in a separate trait and only feature gate on that single trait instead of having all the other requirements repeated.

Reply:
addressed in 5befcaa7ff